src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/extension.ts

```diff
@@ -194,12 +185,23 @@ export async function activate(vscodeType: typeof vscodeapi, context: ExtensionC
         context.subscriptions.push(vscodeType.debug.registerDebugConfigurationProvider('blazorwasm', provider));
 
         languageServerClient.onStarted(async () => {
+            const legend = await languageServiceClient.getSemanticTokenLegend();
+            const semanticTokenProvider = new RazorDocumentSemanticTokensProvider(
+                documentSynchronizer,
+                documentManager,
+                languageServiceClient,
+                logger);
+            if (legend) {
+                localRegistrations.push(vscodeType.languages.registerDocumentSemanticTokensProvider(RazorLanguage.id, semanticTokenProvider, legend));
+                localRegistrations.push(vscodeType.languages.registerDocumentRangeSemanticTokensProvider(RazorLanguage.id, semanticTokenProvider, legend));
+            }
+
             await documentManager.initialize();
         });
 
         await startLanguageServer(vscodeType, languageServerClient, logger, context);
 
-        context.subscriptions.push(languageServerClient, onStartRegistration, onStopRegistration, logger);
```

Comment:
Not sure what the third argument here was, but was it potentially required to ensure `onStopRegistration` is called?

Reply:
I don't follow. Each thing being passed to `push` is an `IDisposable` being added to the list of subscriptions for later disposal. Since `onStartRegistration` doesn't produce a `Disposable` anymore I remove it. I didn't change anything about `onStopRegistration`?

Reply:
Ah, misread the diff. Sounds good, thanks!